test/integration/targets/postgresql/tasks/main.yml

```
@@ -724,6 +724,49 @@
       - "result_database.stdout_lines[-1] == '(1 row)'"
       - "'{{ db_user2 }}=Cc' in result_database.stdout"
 
+- name: Create function for test
+  become: True
+  become_user: "{{ pg_user }}"
+  postgresql_query:
+    login_user: "{{ db_user1 }}"
+    login_password: password
+    db: "{{ db_name }}"
+    query: CREATE FUNCTION public.a() RETURNS integer LANGUAGE SQL AS 'SELECT 2';
+
+- name: Grant execute to all functions
+  postgresql_privs:
+    type: "function"
+    state: "present"
+    privs: "EXECUTE"
+    roles: "{{ db_user2 }}"
+    objs: "ALL_IN_SCHEMA"
+    schema: "public"
+    db: "{{ db_name }}"
+    login: "{{ db_user1 }}"
+    password: "password"
+    host: "localhost"
```

Thank you!

1. First of all, move your tests in a separate file (e.g. postgresql_privs.yml) and then include your file to main.yml in a suitable place.

To make your tests complete:

2. We must be sure that the Grant playbook returns changed == true when it changes, so I suggest
```suggestion
    host: "localhost"
    register: result
    
- assert:
    that:
    - result.changed == true
```
3. Add a playbook that tries to grant the same privs again to check that result.changed == false to be sure that nothing changed.

---

1. Should I move all of the `postgresql_privs` tests or just mine? There are other uses in this `main` file.

---

Just only yours now.
Better to move all in a PR later. But your file will be as a fundamental for that